private/bufpkg/bufconfig/buf_yaml_file.go

```diff
@@ -106,17 +110,35 @@ func NewBufYAMLFile(
 	fileVersion FileVersion,
 	moduleConfigs []ModuleConfig,
 	configuredDepModuleRefs []bufmodule.ModuleRef,
+	options ...BufYAMLFileOption,
 ) (BufYAMLFile, error) {
+	bufYAMLFileOptions := newBufYAMLFileOptions()
+	for _, option := range options {
+		option(bufYAMLFileOptions)
+	}
 	return newBufYAMLFile(
 		fileVersion,
 		nil,
 		moduleConfigs,
 		nil, // Do not set top-level lint config, use only module configs
 		nil, // Do not set top-level breaking config, use only module configs
 		configuredDepModuleRefs,
+		bufYAMLFileOptions.includeDocsLink,
 	)
 }
 
+// BufYAMLFileOption is an option for a new BufYAMLFile
+type BufYAMLFileOption func(*bufYAMLFileOptions)
+
+// BufYAMLFileWithIncludeDocsLink returns a new BufYAMLFileOption that specifies including
+// a comment with a link to the public docs at the top of the buf.yaml file. The comment
+// is included when set to true.
+func BufYAMLFileWithIncludeDocsLink(includeDocsLink bool) BufYAMLFileOption {
```

Comment:
nit: would we prefer `BufYAMLFileWithIncludeDocsLink()` since it's false by default?

Reply:
Yes, that makes a lot of sense to me.